internal/proxy/add.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# add.sh adds new dependencies to the Go Cloud module proxy. This script
+# should be run from the root of the repository.
+
+set -o pipefail
```

Comment:
Check the other bash scripts: we are consistently setting `-e` and a few others.

```suggestion
# https://coderwall.com/p/fkfaqq/safer-bash-scripts-with-set-euxo-pipefail
set -euxo pipefail
```

Reply:
Adding e just in case (not really needed since I check for errors everywhere, but may catch something if someone forgets) and u because that can catch real bugs. I'm leaving out x because I want the script to calmly describe what it's doing in English instead of filling the screen with noise.